src/daemon/core.h

```diff
@@ -78,9 +68,19 @@ class t_core final
 #endif
     if (!m_core.init(m_vm_HACK, nullptr, get_checkpoints))
     {
-      return false;
+      throw std::runtime_error("Failed to initialize core");
```

Comment:
Its not obvious why the code in the `run` function had to be bumped forward to the constructor ... ?

Reply:
This was because RPC was initialized before core. RPC init now uses core (via nettype()).